### PR TITLE
#2107 cash register use cash payment table

### LINKED ISSUE #2107: Update cash register to use new payment types
## Is your feature request related to a problem? Please describe.

See desktop issue https://github.com/sussol/msupply/issues/5346.

## Describe the solution you'd like

- Update incoming sync to handle new payment types table.
- Update cash register modal to use payment options. 

## Implementation

Simplest option is to use existing `Options` table and add a `UIDatabase` case. Other option is to adding a new payment types table. @joshxg @Chris-Petty @andreievg @DhanyaHerath @mark-prins @diegoevangelisti Thoughts?

## Describe alternatives you've considered

N/A.

## Additional context

See epic #1917.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #2107.

## Change summary

Adds payment type functionality to mobile cash register.

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Cash in and cash out transactions cannot be created without specifying a payment type.
- [ ] Transaction payment types are correctly displayed in the cash register.

### Related areas to think about

N/A.

### CODE REVIEW

File: src/database/DataTypes/Transaction.js [hunk 1]
```diff
@@ -171,6 +171,10 @@ export class Transaction extends Realm.Object {
     return (this.option && this.option.title) || '';
   }
 
+  get paymentTypeTitle() {
```
Comment (joshxg) [inline src/database/DataTypes/Transaction.js]: `paymentTypeDescription`?

File: src/utilities/modules/dispensary/constants.js [hunk 2]
```diff
@@ -6,6 +6,7 @@
 export const CASH_TRANSACTION_KEYS = {
   NAME: 'name',
   TYPE: 'title',
+  PAYMENT_TYPE: 'description',
```
Comment (joshxg) [inline src/utilities/modules/dispensary/constants.js]: `DESCRIPTION: 'description'`?

same with the one below, these constants are confusing

Comment (wlthomson) [reply src/utilities/modules/dispensary/constants.js]: Sorry, maybe need a comment. These are the keys for each cash transaction field, i.e. when rendering a `GenericChoiceList` of `PAYMENT_TYPE`, display the `description` field.

Comment (wlthomson) [reply src/utilities/modules/dispensary/constants.js]: 3230de6.

File: src/database/DataTypes/Transaction.js (hunk 1 above)
Comment (wlthomson) [reply src/database/DataTypes/Transaction.js]: 54525f2.

**Pull request merged**